### PR TITLE
Draft Autoswap PR

### PR DESCRIPTION
**Pull request by eljhfx**
Teddy asked for a  draft PR to review

### CODE REVIEW

File: proto/dex/tx.proto [hunk 1]
```diff
@@ -36,7 +36,7 @@ message MsgDeposit {
   ]; 
   repeated int64 tickIndexes = 7;
   repeated uint64 feeIndexes = 8;
-  
+  bool autoswap = 9;
```
Comment (teddyknox) [inline proto/dex/tx.proto:39]: can we put this within a DepositOptions field?

File: x/dex/keeper/core.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ func (k Keeper) DepositCore(
 	callerAddr sdk.AccAddress,
 	amounts0 []sdk.Int,
 	amounts1 []sdk.Int,
+	autoswap bool,
```
Comment (teddyknox) [inline x/dex/keeper/core.go:28]: autoswap is already on msg, no need to pass this in

File: x/dex/keeper/core.go [hunk 3]
```diff
@@ -85,7 +86,7 @@ func (k Keeper) DepositCore(
 		oldReserve0 := pool.GetLowerReserve0()
 		oldReserve1 := pool.GetUpperReserve1()
 
-		inAmount0, inAmount1, outShares := pool.Deposit(amount0, amount1, totalShares)
+		inAmount0, inAmount1, outShares := pool.Deposit(amount0, amount1, totalShares, autoswap)
```
Comment (teddyknox) [inline x/dex/keeper/core.go:89]: pass in options here

File: x/dex/keeper/msg_server.go [hunk 4]
```diff
@@ -23,12 +23,13 @@ var _ types.MsgServer = msgServer{}
 func (k msgServer) Deposit(goCtx context.Context, msg *types.MsgDeposit) (*types.MsgDepositResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	token0, token1, createrAddr, amount0, amount1, err := k.DepositVerification(goCtx, *msg)
+	token0, token1, autoswap, createrAddr, amount0, amount1, err := k.DepositVerification(goCtx, *msg)
```
Comment (teddyknox) [inline x/dex/keeper/msg_server.go:26]: why does this need to be returned?

File: proto/dex/tx.proto (hunk 1 above)
Comment (eljhfx) [reply proto/dex/tx.proto:39]: Not sure what a DepositsOptions field is but assume it makes it an optional parameter (defaulted to false?). Think this is a good change if yeah.

File: x/dex/keeper/core.go (hunk 2 above)
Comment (eljhfx) [reply x/dex/keeper/core.go:28]: Ahhhh I see now. When I saw token0,1 and the other fields there I replicated it with autoswap but realize now that this was just bc they need to be sorted or the type needs to change.

File: x/dex/keeper/msg_server.go [hunk 5]
```diff
@@ -23,12 +23,13 @@ var _ types.MsgServer = msgServer{}
...
 
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO: DECIDE WHETHER WE SHOULD HAVE A SEPERATE MSG SERVER FUNCTION FOR DEPOSIT
```
[the remark this answers is not in the document]
Comment (eljhfx) [reply x/dex/keeper/msg_server.go:32]: Agreed

File: x/dex/keeper/msg_server.go (hunk 4 above)
Comment (eljhfx) [reply x/dex/keeper/msg_server.go:26]: This was from when autoswap was a param in `DepositCore`, but changing it since it's unnecessary

**Pull request closed without merging**